src/__tests__/real-world-example/real-world-example.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { getPage } from '../../index';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import path from 'path';
+import userEvent from '@testing-library/user-event';
+
+describe('real-world-example', () => {
+  it('Should work as expected', async () => {
+    const { page } = await getPage({
+      nextRoot: path.join(__dirname, '__fixtures__'),
+      route: '/?name=Matthew',
+      useDocument: true,
+    });
+
+    const { container } = render(page);
+
+    const head = container.querySelector('head') as HTMLHeadElement;
+    const html = container.querySelector('html') as HTMLHtmlElement;
+
+    expect(html).toHaveAttribute('lang', 'en');
+    expect(head.querySelector('meta[name="Description"]')).toHaveAttribute(
+      'Content',
+      'Custom document description'
+    );
+
+    expect(head.querySelector('title')?.textContent).toEqual('Create Next App');
+
+    // Correctly passes query
+    expect(screen.getByText('Hello Matthew')).toBeInTheDocument();
+
+    userEvent.click(screen.getByText('To page A'));
+
+    // TODO: fix client side navigation when useDocument: true
+    // await screen.findByText('Back to root');
```

Comment:
Client side navigation does not work when `useDocument: true` -- TODO: investigate

Reply:
I haven't had time to verify, but I guess this could be at least partially related to the way we wrap the Pafe (including `Document` which is written to be rendered server-side, only) with `RouterProvider` and `NavigationProvider`.

Client navigation is currently implemented as follows:
- `RouterProvider` wraps the page and makes available a the router mock for current page via context
- router mock is configured to issue a router mock update on `route.push`/`route.replace`
- Below `RouterProvider` lives `NavigationProvider` which invokes `makePage` each time router object changes

`makePage` creates a new `pageObject` based on the new router and calls `makePageElement` which recreates the whole Document + App + Page element from scratch.

All this happens on React runtime.

Reply:
Okay, I have come up with a way to make it work -- more hacks :)

The problem is that as soon as you render elements on server using `renderToStatic*` function scope is lost it won't work as expected in the tests.

Will prepare changes later today.

Reply:
Should be good to go @toomuchdesign 🚀